### PR TITLE
ENG-77-Proration - tests

### PR DESCRIPTION
**Pull request by Connorbelez**
## Summary by CodeRabbit

* **Tests**
  * Enhanced proration test coverage with comprehensive integration test suite.
  * Added validation for proration calculations across multiple scenarios, including partial transfers and month-boundary edge cases.
  * Expanded testing to verify accrual accuracy and consistency across complex transaction conditions.

### CODE REVIEW

**Comment by Connorbelez** [comment]
* **#143** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/143?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/143?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: convex/accrual/__tests__/proration.test.ts [hunk 1]
```diff
@@ -1,101 +1,646 @@
...
+		// Invariant: seller + buyer = single owner
+		const buyerAccrual = calculateAccrualForPeriods(
+			buyerPeriods,
+			RATE,
+			PRINCIPAL,
+			"2026-01-01",
+			"2026-01-31"
+		);
+		const singleOwner = calculatePeriodAccrual(RATE, 1, PRINCIPAL, 31);
+		expect(sellerAccrual + buyerAccrual).toBe(singleOwner);
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/accrual/__tests__/proration.test.ts]: Fixed: Changed toBe to toBeCloseTo for floating-point invariant assertions.

File: convex/accrual/__tests__/proration.test.ts [hunk 2]
```diff
@@ -1,101 +1,646 @@
...
+			RATE,
+			PRINCIPAL,
+			"2026-01-01",
+			"2026-01-31"
+		);
+		expect(buyerAccrual).toBe(0);
+
+		// Invariant still holds: seller + buyer = single owner (31 days)
+		const singleOwner = calculatePeriodAccrual(RATE, 1, PRINCIPAL, 31);
+		expect(sellerAccrual + buyerAccrual).toBe(singleOwner);
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/accrual/__tests__/proration.test.ts]: Fixed: Changed toBe(0) to toBeCloseTo(0, 5) and toBe(singleOwner) to toBeCloseTo(singleOwner, 5) for floating-point safety.

File: convex/accrual/__tests__/proration.test.ts [hunk 3]
```diff
@@ -1,101 +1,646 @@
...
+// ---------------------------------------------------------------------------
+
+describe("invariant: seller_accrual + buyer_accrual = single_owner_accrual", () => {
+	// Transfer dates covering various month positions
+	const transferDates = [
+		"2026-01-05", // early month
+		"2026-01-15", // mid month
+		"2026-01-20", // late month
+		"2026-01-28", // last week
+		"2026-02-15", // leap year Feb
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/accrual/__tests__/proration.test.ts]: Fixed: Corrected comment — 2026 is not a leap year (not divisible by 4), changed 'leap year Feb' to 'mid-February'.

File: convex/accrual/__tests__/proration.test.ts [hunk 4]
```diff
@@ -1,101 +1,646 @@
...
+		"2026-04-30", // 30-day month
+		"2026-12-31", // last day of year
+	];
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/accrual/__tests__/proration.test.ts]: Fixed: Corrected comment — 2026 is not a leap year, changed 'leap year Feb' to 'mid-February' as suggested.

File: convex/accrual/__tests__/proration.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/accrual/__tests__/proration.test.ts]: Verified fixed in commit.
